### ISSUE TITLE
E2E Not testing stripe subscription billing 

### ISSUE THREAD

**Issue by brucealdridge** [Automattic/woocommerce-payments#5041]
### Describe the bug

WooCommerce Subscriptions is installed as part of the E2E test suite. Because WC Subscriptions is always running and active for all E2E tests, the flows are all tested using tokenised subscriptions. 

We should also be testing stripe billing subscriptions which are used when WC Subscriptions is not installed or activated.

**Comment by Jinksi** [comment]
Currently, we have a single [`subscriptions` e2e test group](https://github.com/Automattic/woocommerce-payments/blob/eb7118da4735624e21a3f6918258a36663d38269/tests/e2e/specs/subscriptions) with `merchant` and `shopper` branches.

We could have 2 subscription e2e test groups, something like:
```
subscriptions-tokenised
- merchant
- shopper
subscriptions-stripe-billing
- merchant
- shopper
```

~I'm assuming that~ Most (12/15) of the test flows are shared across tokenised/stripe-billing, in which case there would be too much duplication with this method.

**Update**: I've performed a quick review of the existing tests to determine how many are compatible with WCPay Subs. It looks like 12/15 existing tests _should_ work for both WC Subs & WCPay Subs.

<details>
<summary>Existing tests compatibility table</summary>

| File                                                          | Test Description                                                                  | WC Subs | WCPay Subs |
| ------------------------------------------------------------- | --------------------------------------------------------------------------------- | ------- | ---------- |
| merchant-subscriptions-renew-action-scheduler.spec.js         | 'should be able to renew a subscription via Action Scheduler'                     | x       |            |
| merchant-subscriptions-renew-subscription.spec.js             | 'should be able to renew a subscription as a merchant'                            | x       |            |
| merchant-subscriptions-renew-subscription.spec.js             | 'should be able to load WooCommerce Subscriptions Settings tab'                   | x       | x          |
| shopper-myaccount-renew-subscription.spec.js                  | 'should be able to purchase a subscription'                                       | x       | x          |
| shopper-myaccount-renew-subscription.spec.js                  | 'should be able to renew a subscription in my account'                            | x       |            |
| shopper-subscriptions-manage-payments.spec.js                 | 'should be able to purchase a subscription'                                       | x       | x          |
| shopper-subscriptions-manage-payments.spec.js                 | 'should change a default payment method to a new one'                             | x       | x          |
| shopper-subscriptions-manage-payments.spec.js                 | 'should set a payment method to an already saved card'                            | x       | x          |
| shopper-subscriptions-purchase-free-trial.spec.js             | 'should be able to purchase a subscription with free trial'                       | x       | x          |
| shopper-subscriptions-purchase-free-trial.spec.js             | 'should create an order with "Setup Intent"'                                      | x       | x          |
| shopper-subscriptions-purchase-multiple-subscriptions.spec.js | 'should be able to purchase multiple subscriptions'                               | x       | x          |
| shopper-subscriptions-purchase-no-sign-up-fee.spec.js         | 'should be able to purchase a subscription with signup fee'                       | x       | x          |
| shopper-subscriptions-purchase-no-sign-up-fee.spec.js         | 'should have a charge for subscription cost without fee & an active subscription' | x       | x          |
| shopper-subscriptions-purchase-sign-up-fee.spec.js            | 'should be able to purchase a subscription with signup fee'                       | x       | x          |
| shopper-subscriptions-purchase-sign-up-fee.spec.js            | 'should have a charge for subscription cost with fee & an active subscription'    | x       | x          |

</details>

**Comment by Jinksi** [comment]
We could activate/deactivate WC Subscriptions before/after tests, in a similar way to the [`activateWooPay()`](https://github.com/Automattic/woocommerce-payments/blob/eb7118da4735624e21a3f6918258a36663d38269/tests/e2e/utils/flows.js#L498)/[`deactivateWooPay()`](https://github.com/Automattic/woocommerce-payments/blob/eb7118da4735624e21a3f6918258a36663d38269/tests/e2e/utils/flows.js#L529) or [`activateUpe()`](https://github.com/Automattic/woocommerce-payments/blob/eb7118da4735624e21a3f6918258a36663d38269/tests/e2e/utils/flows.js#L257)/[`deactivateUpe()`](https://github.com/Automattic/woocommerce-payments/blob/eb7118da4735624e21a3f6918258a36663d38269/tests/e2e/utils/flows.js#L284).

This might slow down tests somewhat, but it would be faster than teardown/setup of a new environment. (Although GH actions can run in parallel, our local dev env cannot AFAIK).

**Update:** using this method, I'm seeing the following activation/deactivation times when running locally in interactive mode (`e2e-dev`):
* Activating WCS: 6.3s
* Deactivating WCS: 5.4s

<details>
<summary>Code example</summary>

```js
// tests/e2e/utils/flows.js
export const merchantWCP = {
...
	activateWCS: async () => {
		const start = Date.now();
		await merchant.openPlugins();
	
		if ( await page.$( '#activate-woocommerce-subscriptions' ) ) {
			console.log( 'Activating WCS' );
			await expect( page ).toClick(
				'#activate-woocommerce-subscriptions'
			);
			await page.waitForNavigation( {
				waitUntil: 'networkidle0',
			} );
		} else {
			console.log( 'WCS already activated' );
		}
	
		// Confirm activation
		await page.$( '#deactivate-woocommerce-subscriptions' );
		const end = Date.now();
		console.log( `WCS activated in ${ end - start }ms` );
	},
	
	deactivateWCS: async () => {
		const start = Date.now();
		await merchant.openPlugins();
	
		if ( await page.$( '#deactivate-woocommerce-subscriptions' ) ) {
			console.log( 'Deactivating WCS' );
			await expect( page ).toClick(
				'#deactivate-woocommerce-subscriptions'
			);
			// Modal dialog appears, click "Deactivate" button.
			await expect( page ).toMatchElement( 'h1', {
				text: 'Are you sure?',
			} );
			await expect( page ).toClick(
				'button#wcpay-subscriptions-plugin-deactivation-submit'
			);
			await page.waitForNavigation( {
				waitUntil: 'networkidle0',
			} );
		} else {
			console.log( 'WCS already deactivated' );
		}
	
		// Confirm deactivation
		await page.$( '#activate-woocommerce-subscriptions' );
		const end = Date.now();
		console.log( `WCS deactivated in ${ end - start }ms` );
	},
}
```
</details>

**Comment by haszari** [comment]
Marking as high priority. Rationale:

- Our e2e tests are testing the wrong code (we need tests that test WCPay / Stripe Billing subscriptions).
- We should be able to fix by tweaking the test script or environment (so WC S is not active).

**Comment by frosso** [comment]
Closing — WCPay Subscriptions (Stripe Billing) was deprecated in Feb 2025, so adding E2E test coverage for the Stripe Billing subscription flow is no longer needed.
